### ISSUE TITLE
Renew changes common name

### ISSUE THREAD

**Issue by mjgtall** [certbot/certbot#3127]
I created a multi-domain certificate using certbot certonly -d mydomain -d a.mydomain -d b.mydomain which had the common name mydomain. After running certbot renew it renewed the certificate but made a.mydomain the common name.

I don't know if this is a blocker for any use case, it certainly works for me. But I would like the renewed certificates to keep the old common name. It especially makes it easier to work with multiple certificates if you can map the common name of the certificate to the entry in /etc/letsencrypt/live/.

**Comment by bmw** [comment]
Do you have any logs from the renewal run that changed the common name? They're in `/var/log/letsencrypt` by default.

Additionally, what is the output of `ls -l /etc/letsencrypt/archive/* /etc/letsencrypt/live/*`?

Feel free to redact domain names and potentially e-mail addresses if you want.

**Comment by marcusk** [comment]
I'm also running into this issue. I Initially created a certificate like this using certonly:

`certbot certonly -a webroot -w /var/conf/letsencrypt -d www.acc.umu.se -d autoconfig.acc.umu.se -d git.acc.umu.se -d mattith.acc.umu.se -d puss.acc.umu.se -d sms.acc.umu.se -d temp.acc.umu.se -d webmail.acc.umu.se -d www-ntk.acc.umu.se -d www.ntk.umu.se`

I then forced a renewal using `certbot renew --force-renewal`. In the initial certificate the common name was `www.acc.umu.se`, after renewal it became `autoconfig.acc.umu.se`:
```
root@mao:~# openssl x509 -noout -subject -in /etc/letsencrypt/archive/www.acc.umu.se/cert1.pem
subject= /CN=www.acc.umu.se
root@mao:~# openssl x509 -noout -subject -in /etc/letsencrypt/archive/www.acc.umu.se/cert2.pem
subject= /CN=autoconfig.acc.umu.se
```

Attaching [letsencrypt.log.1](https://github.com/certbot/certbot/files/301374/letsencrypt.log.1.txt) from creating the certificate and also [letsencrypt.log](https://github.com/certbot/certbot/files/301361/letsencrypt.log.txt) from the renewal.

Output from ls:
```
/etc/letsencrypt/archive/www.acc.umu.se:
total 32
-rw-r--r-- 1 root root 2021 Jun  6 23:12 cert1.pem
-rw-r--r-- 1 root root 2033 Jun  6 23:14 cert2.pem
-rw-r--r-- 1 root root 1647 Jun  6 23:12 chain1.pem
-rw-r--r-- 1 root root 1647 Jun  6 23:14 chain2.pem
-rw-r--r-- 1 root root 3668 Jun  6 23:12 fullchain1.pem
-rw-r--r-- 1 root root 3680 Jun  6 23:14 fullchain2.pem
-rw-r--r-- 1 root root 1704 Jun  6 23:12 privkey1.pem
-rw-r--r-- 1 root root 1704 Jun  6 23:14 privkey2.pem


/etc/letsencrypt/live/www.acc.umu.se:
total 0
lrwxrwxrwx 1 root root 38 Jun  6 23:14 cert.pem -> ../../archive/www.acc.umu.se/cert2.pem
lrwxrwxrwx 1 root root 39 Jun  6 23:14 chain.pem -> ../../archive/www.acc.umu.se/chain2.pem
lrwxrwxrwx 1 root root 43 Jun  6 23:14 fullchain.pem -> ../../archive/www.acc.umu.se/fullchain2.pem
lrwxrwxrwx 1 root root 41 Jun  6 23:14 privkey.pem -> ../../archive/www.acc.umu.se/privkey2.pem
```

**Comment by mjgtall** [comment]
Sorry, I don't have the logs anymore but what @marcusk writes is exactly the same as I did. When creating the certificate the first domain given to the certonly command using the -d option became the CN, after renewal the CN became the first domain when sorted alphabetically.

**Comment by bmw** [comment]
Do you know if you were using version 0.5.0 of the `letsencrypt` client when `renew` changed the CN in your cert? There was a regression in that version of the client that would have caused this behavior.

If this is the problem, it has been fixed in all newer versions of the client (see #2798). Assuming that was the problem, to fix your domain order, renew once using the `certonly` command, specifying all domains on the command line with the first domain being the one that should be the CN. After that, `renew` should work.

If you weren't using version 0.5.0, however, I have some more digging to do.

**Comment by marcusk** [comment]
I'm able to reproduce this with 0.8.0 at both certonly and renew time, installed in a blank virtualenv via pip install certbot.

**Comment by jaimbo** [comment]
I am experiencing the same issue. Renew has changed my common name to the first result alphabetically.

**Comment by bmw** [comment]
Just repoed this.

We'll get this fixed for the next release.

**Comment by dusek** [comment]
Just wondering if the change in CN (though no change in SAN set) is treated as a new FQDN set for the purposes of [rate limits](https://community.letsencrypt.org/t/rate-limits-for-lets-encrypt/6769)?

**Comment by bmw** [comment]
@dusek, no it will still be treated as the same FQDN set.

**Comment by jagerman** [comment]
Just a note for anyone coming across this: while this is fixed in 0.8.1, the fix there only preserves whatever CN is currently in the certificate.  Unfortunately this means that if a certificate renewal changed the CN, renewal will now try to keep the already-changed name.  To fix: run `certbot` with the arguments necessary to create the certificate as a new certificate (list the desired CN as the first domain), and when asked, say that you really want a new one despite the set of domains being the same as an already-valid certificate.  The new one will have the desired CN, and future renewals will maintain it.
